device/config/configurator.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"github.com/xshellinc/tools/dialogs"
+	"github.com/xshellinc/tools/lib/ssh_helper"
+)
+
+// Default Configurator constants that are describe a specific configuration option
+const (
+	Locale = iota
+	Keymap
+	Wifi
+	Interface
+	Dns
+)
+
+// consts are a const strings representation
+var consts = [...]string{
+	"Locale",
+	"Keymap",
+	"Wifi",
+	"Interface",
+	"Dns",
+}
+
+// GetConstLiteral gets a literal from configurator.const
+func GetConstLiteral(v int) string {
+	return consts[v]
+}
+
+type (
+	// Configurator is a config helper, which uses ConfigCallbackFn to store configurations for devices
+	Configurator interface {
+		Setup() error
+		Write() error
+		SetConfigFn(int, *ConfigCallbackFn)
+		GetConfigFn(int) *ConfigCallbackFn
+	}
+
+	// configurator is a container of a mutual storage and order of ConfigCallbackFn
+	configurator struct {
+		storage map[string]interface{}
+		order   []*ConfigCallbackFn
+	}
+
+	// ConfigFunction is a function with an input parameter of configurator's `storage`
+	ConfigFunction func(map[string]interface{}) error
```

Comment:
__[golint]__
* [golint] type name will be used as config.ConfigFunction by other packages, and that stutters; consider calling this Function

<!-- Comment by [SideCI](https://sideci.com) -->